students/clifford_butler/lesson04/mailroom2.py

```diff
@@ -0,0 +1,148 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Sun Mar 29 16:04:35 2020
+
+@author: cliff
+"""
+
+#!/usr/bin/env python3
+'''
+Use dicts where appropriate.
+Part 1 of this assignment used these basic data types: numbers, strings, 
+ists and tuples.
+
+However, using dictionaries, covered in Lesson 4, will let you re-write 
+your program a bit more simply and efficiently.
+
+Update your mailroom program to:
+
+Use dicts where appropriate.
+See if you can use a dict to switch between the user’s selections.
+See if you can use a dict to switch between the users selections. 
+see Using a Dictionary to switch for what this means.
+Convert your main donor data structure to be a dict.
+Try to use a dict and the .format() method to produce the letter 
+as one big template, rather than building up a big string that 
+produces the letter in parts.
+'''
+
+import sys
+import operator
+
+donor_dict = {"William Gates, III": [653772.32, 12.17],
+        "Jeff Bezos":  [877.33],
+        "Paul Allen": [663.23, 43.87, 1.32],
+        "Mark Zuckerberg": [1663.23, 4300.87, 10432.0],
+        "Alexandra Butler": [777.77, 44.44]}
+
+prompt = "\n".join(("Welcome to the mail room!",
+          "Please choose from below options:",
+          "1 - Send a Thank You",
+          "2 - Create a Report",
+          "3 - Send letters to all donors",
+          "4 - Quit",
+          ">>> "))
+
+prompt_name = "\n".join(("Type the donors full name or,",
+              "type 'list' to display a list of the donors names.",
+              ">>> "))
+
+prompt_amount = "\n".join(("Whats the donation amount?",
+                ">>> "))
+
+exit_report = "\n".join(("Press 1 to exit to the initial prompt.",
+                         "\n"))
+
+def get_index(donor_name):
+    # Return the index number based on user input
+    for item in (donor_dict):
+        if item[0] == donor_name:
+            index = donor_dict.index(item)
+            
+            return index
+        
+    return None
+
+def add_donor(donor_name):
+    # Add donor name to the data structure.
+    for item in donor_dict:
+        if item[0] not in donor_dict:
+            donor_dict[donor_name] = []
+            print (donor_dict)
+            break
```

Comment:
looks like get_index and add_donor are leftover functions from list implementation?